### ISSUE TITLE
Bug: inconsistency for the BatchNormalization layer between Tensorflow, CNTK and Theano backends

### ISSUE THREAD

**Issue by Dandy-John** [keras-team/keras#12957]
### Configuration

| Library                        | Version  |
| ------------------------------ | -------- |
| python                         | 3.6.8    |
| GCC                            | 7.3.0    |
| tensorflow-base/tensorflow-gpu | 1.13.1   |
| keras-gpu/keras-base           | 2.2.4    |
| theano                         | 1.0.4    |
| cudnn                          | 7.3.1    |
| cudatoolkit                    | 10.0.130 |

Machine Configuration：

Ubuntu 18.04.2 LTS

Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz 128GB RAM

CUDA Version: 10.2 

Double GeForce GTX 1080ti 

---

### Description

This bug happened when I was using a pre-trained keras model called Xception. I loaded the same model on different backends(tensorflow, CNTK and theano). When I tried to do some prediction, I found that the prediction accuracy of tensorflow are different from that of CNTK and theano. Among 1500 images, the number of images for which tensorflow and CNTK have different results is 118, the number of images for which tensorflow and theano have different results is also 118, which theano and CNTK  do not have any different result.

Some of the prediction results of this model on 3 backends are shown as below.

![result](https://user-images.githubusercontent.com/17698785/59504767-3ed80200-8ed6-11e9-8025-5f7e8246ef5e.png)

It seems that there is something wrong with tensorflow. I put images into the model and recorded the ouputs of each layers to see how this happens. It seems that the outputs on CNTK and theano are very close, but for tensorflow, the outputs of a BatchNormalization layer and all the layers after this layer are very different from CNTK and theano. 

The results are attached below.

#### comparation of tensorflow and cntk
![tf_cn](https://user-images.githubusercontent.com/17698785/59504976-c9206600-8ed6-11e9-97b1-ba0e8b4e86e8.png)

#### comparation of tensorflow and theano
![tf_th](https://user-images.githubusercontent.com/17698785/59505044-e6553480-8ed6-11e9-8c73-6daa6a3bb4be.png)

#### comparation of cntk and theano
![th_cn](https://user-images.githubusercontent.com/17698785/59505080-fcfb8b80-8ed6-11e9-97b4-179036e2d48b.png)

The delta calumn shows the discrepancy, which is calculated as:
![image](https://user-images.githubusercontent.com/17698785/59505143-27e5df80-8ed7-11e9-869c-5066fb93af50.png)

Based on the results, the deviation seems to be caused by batch_normalization_2 layer of tensorflow backend.

**Comment by ymodak** [comment]
n order to expedite the trouble-shooting process, can you please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by Dandy-John** [comment]
Here is the code:
[code.zip](https://github.com/keras-team/keras/files/3299973/code.zip)

1. Make output directory and localize_res directory

2. Run get_layers_output.py on different backends with some image
   Please run this script on tensorflow, theano, and cntk
   e.g. python get_layers_output.py tensorflow 116_corkscrew.png
   
3. Run get_layers_deviation.py with some image
   e.g. python get_layers_deviation.py 116_corkscrew.png
   
4. Result can be found in localize_res directory

**Comment by benjamin-lucas** [comment]
I've had a lot of issues with the bath_norm layer for transfer learning too. There is a bug using the TF backend where the initial values of the moving_mean and moving_var are ignored. This is likely (one of) the cause(s) of the inconsistency.

I've raised it here: [https://github.com/keras-team/keras/issues/13214](url)
